### DISCUSSION TITLE
GDB warnings about variables in J-link debug

### DISCUSSION THREAD

**Discussion by Yuhang-Yohoo** [bitcraze/discussions#1511] [Q&A]
Hi devs, here I met some problems when I was debugging with Jlink in VScode, WSL2, windows
![image](https://github.com/user-attachments/assets/0ba981ce-fc33-4661-b6e6-62f7e72243ed)
The problem is: once I start debugging,(if I drag my cursor to a variable) the debug console keeps showing GDB warning messages about the variables in my code. Also I cannot add variables to the watch window. I searched for these messages and it told me this is due to the compiler optimization, I tied to add "ccflags-y += -O0 -g" in the root Makefile but it doesn't work, any suggestions will be appreciated.

**Comment by Yuhang-Yohoo** [comment]
Some of the warnings may look like:

```
GDB -> App: {"token":975,"outOfBandRecord":[],"resultRecords":{"resultClass":"done","results":[["changelist",[]]]}}
GDB -> App: {"token":976,"outOfBandRecord":[],"resultRecords":{"resultClass":"done","results":[["changelist",[]]]}}
GDB -> App: {"token":977,"outOfBandRecord":[],"resultRecords":{"resultClass":"error","results":[["msg","Variable object not found"]]}}
979-var-create watch_bece5f83e0148e19a6a88a66251b9d90182a8843d9c281bbff8faa12027d4c9c @ "pDAta"
GDB -> App: {"token":978,"outOfBandRecord":[],"resultRecords":{"resultClass":"done","results":[["changelist",[[["name","watch_c9a57ee8a098d854c28637e58e20cc54f092604b21e6efb71896f6b58fadaecd"],["value","0x0"],["in_scope","true"],["type_changed","false"],["has_more","0"]]]]]}}
GDB -> App: {"token":979,"outOfBandRecord":[],"resultRecords":{"resultClass":"error","results":[["msg","-var-create: unable to create variable object"]]}}
watch: -var-create: unable to create variable object (from var-create watch_bece5f83e0148e19a6a88a66251b9d90182a8843d9c281bbff8faa12027d4c9c @ "pDAta")980-stack-list-variables --thread 1 --frame 0 --simple-values
GDB -> App: {"token":980,"outOfBandRecord":[],"resultRecords":{"resultClass":"done","results":[["variables",[[["name","this"],["arg","1"],["type","OutlierFilterBuffer_t *"],["value","0x0"]]]]]}}
```

**Comment by knmcguire** [comment]
hi! we haven't used WSL2 with the gdb debugging. What are you using to connect the usb to your WSL2? USBipd?

**Reply by Yuhang-Yohoo** [reply]
> hi! we haven't used WSL2 with the gdb debugging. What are you using to connect the usb to your WSL2? USBipd?
Thanks for the reply.

Yes I followed your debugging instructions like:
` usbipd attach -b 3-3 --wsl`

The debugger I used is the newest Jlink, but I don't understand why there are so many GDB warnings, actually they don't impact the work, just keep showing up during the debug, it's annoying.

**Reply by knmcguire** [reply]
hi and sorry for the late reply. We have been able to confirm this issue but we are having trouble trying to figure out what is really going on. Technically the j[link is not on usbipd's tested devices](https://github.com/dorssel/usbipd-win/wiki/Tested-Devices) and only the stlink is, but I and @ataffanel can't really see why it couldn't work....

We will continue look at this, but I've made an issue for it: https://github.com/bitcraze/crazyflie-firmware/issues/1402

**Reply by Yuhang-Yohoo** [reply]
> hi and sorry for the late reply. We have been able to confirm this issue but we are having trouble trying to figure out what is really going on. Technically the j[link is not on usbipd's tested devices](https://github.com/dorssel/usbipd-win/wiki/Tested-Devices) and only the stlink is, but I and @ataffanel can't really see why it couldn't work....
> 
> We will continue look at this, but I've made an issue for it: [bitcraze/crazyflie-firmware#1402](https://github.com/bitcraze/crazyflie-firmware/issues/1402)

No problem, these warnings actually never obstruct the debugger anyway.
